applications/FluidDynamicsApplication/custom_processes/surface_smoothing_process.h

```diff
@@ -0,0 +1,232 @@
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) SurfaceSmoothingProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of SurfaceSmoothingProcess
+    KRATOS_CLASS_POINTER_DEFINITION(SurfaceSmoothingProcess);
+
+    typedef UblasSpace<double, CompressedMatrix, Vector> TSparseSpace;
+    typedef UblasSpace<double, Matrix, Vector> TDenseSpace;
+    typedef LinearSolver<TSparseSpace, TDenseSpace > TLinearSolver;
+    typedef Scheme< TSparseSpace,  TDenseSpace > SchemeType;
+    typedef typename SchemeType::Pointer SchemePointerType;
+    typedef typename BuilderAndSolver<TSparseSpace,TDenseSpace,TLinearSolver>::Pointer BuilderSolverPointerType;
+    typedef SolvingStrategy< TSparseSpace, TDenseSpace, TLinearSolver > SolvingStrategyType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    SurfaceSmoothingProcess(
+        ModelPart& rModelPart,
+        TLinearSolver::Pointer);
+
+    /// Constructor with Kratos parameters.
+    SurfaceSmoothingProcess(
+        ModelPart& rModelPart,
+        Parameters& rParameters);
+
+    /// Constructor with Kratos model
+    SurfaceSmoothingProcess(
+        Model& rModel,
+        Parameters& rParameters);
+
+    /// Destructor.
+    ~SurfaceSmoothingProcess() override
+    {
+        Model& current_model = mrModelPart.GetModel();
+        if(current_model.HasModelPart( mAuxModelPartName ))
+            current_model.DeleteModelPart( mAuxModelPartName );
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void Execute() override;
+
+    void ExecuteInitialize() override;
+
+    void ExecuteBeforeSolutionLoop() override;
+
+    void ExecuteInitializeSolutionStep() override;
+
+    void ExecuteFinalizeSolutionStep() override;
+
+    void Clear();
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "Surface Smoothing Process" ;
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override {rOStream << "Surface Smoothing Process";}
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrModelPart;
+
+    std::string mAuxModelPartName = "Aux_Smoothing_Model_Part";
+
+    SolvingStrategyType::UniquePointer mp_solving_strategy;
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void InitializeSolutionStrategy(
```

Comment:
The method name is misleading. The `Initialize` function of the solution strategy is not called inside. I suggest renaming it to `CreateSolutionStrategy`.

Reply:
Done